src/Migrations/M034MigrateSplit2ColumnBlock.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+declare(strict_types=1);
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use WP_Block_Parser;
+
+/**
+ * Migrate split-two-columns block to columns block.
+ */
+class M034MigrateSplit2ColumnBlock extends MigrationScript
+{
+    /**
+     * Extract split-two-columns block from page/posts and transform it into columns block.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    public static function execute(MigrationRecord $record): void
+    {
+        try {
+            // Get the list of posts using split-two-columns block.
+            $posts = Utils\Functions::get_posts_using_specific_block(
+                Utils\Constants::BLOCK_SPLIT_TWO_COLUMNS,
+                Utils\Constants::ALL_POST_TYPES
+            );
+
+            // If there are no posts, abort.
+            if (!$posts) {
+                return;
+            }
+
+            $parser = new WP_Block_Parser();
+
+            echo "Split 2 Columns block migration in progress...\n"; // phpcs:ignore
+
+            foreach ($posts as $post) {
+                if (empty($post->post_content)) {
+                    continue;
+                }
+
+                $current_post_id = $post->ID; // Store the current post ID
+
+                echo 'Parsing post ', $post->ID, "\n"; // phpcs:ignore
+                //$result = false;
+
+                // Go through the post blocks to find the planet4-blocks/split-two-columns one.
+                $blocks = $parser->parse($post->post_content);
```

Comment:
We could add an additional control here:

```suggestion
                $blocks = $parser->parse($post->post_content);
                if (!is_array($blocks)) {
                    throw new \Exception("Invalid block structure for post #" . $current_post_id);
                }
```